_rules/text-spacing-not-important-b8bb68.md

```diff
@@ -0,0 +1,287 @@
+---
+id: b8bb68
+name: Text spacing in `style` attributes is not `!important`
+rule_type: atomic
+description: |
+  This rule checks that text spacing set in the `style` attribute is below permissible minimum value or not set to `!important`
+accessibility_requirements:
+  wcag21:1.4.12: # Text Spacing (AA)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+acknowledgments:
+  authors:
+    - Jey Nandakumar
+---
+
+## Applicability
+
+This rule applies to any HTML element that is [visible][] and has one of the following CSS properties set by the `style` attribute:
+
+- [word-spacing][]
+- [letter-spacing][]
+- [line-height][]
+
+## Expectation 1
+
+The test target does not have the [important flag][] set for the CSS [word-spacing][] property, unless the [computed][] [word-spacing][] is greater than `0.16` times the [computed][] [font-size][] of the test target.
+
+## Expectation 2
+
+The test target does not have the [important flag][] set for the CSS [letter-spacing][] property, unless the [computed][] [letter-spacing][] is greater than `0.12` times the [computed][] [font-size][] of the test target.
+
+## Expectation 3
+
+The test target does not have the [important flag][] set for the CSS [line-height][] property, unless the [computed][] [line-height][] is greater than `1.5` times the [computed][] [font-size][] of the test target.
```

Comment:
This seems pretty close! It's still not clear _which_ `line-height` property we're referring to in reference to the important flag. The earliest [value stage](https://drafts.csswg.org/css-cascade/#value-stages) that makes sense in this context would probably be the [cascaded value](https://drafts.csswg.org/css-cascade/#cascaded), but the [computed value](https://drafts.csswg.org/css-cascade/#computed) seems fine as well.

Reply:
@kasperisager I read up the value stages & I am inclined to keep this as computed value. I do not want to mix this with cascaded value as the other 2 expectations use the computed value & also thinking from an implementation perspective, it would make sense to leave this rely on the computed value.

Thoughts welcome. Ta.

Reply:
To be clear, it's this bit I'm referring to:

> The test target does not have the important flag set for the CSS line-height property.

There are multiple different ways to interpret "line-height property" here. It could be the declared property, of which there might be several. It could also be the cascaded property, of which there's only ever 1.

Reply:
@kasperisager I see your point, but the [applicability][https://github.com/act-rules/act-rules.github.io/pull/1251#discussion_r440244173] of the rule, only applies to what is specified in the `style` attribute of the given element.

Reply:
That doesn't make the expectations any less ambiguous though.

Reply:
@kasperisager what do you think of this expectation for the line height:

`The test target does not have the [important flag][] set for the [declared value][https://drafts.csswg.org/css-cascade/#declared] of the CSS [computed][] [line-height][] property, unless the [computed][] [line-height][] is at least `1.5` times the [computed][] [font-size][] of the test target.
`

The thinking is given the applicability is for the style attribute specified for the element. It would make sense to use the declared value over the cascaded value.

Reply:
We're progressing! I think you're right that the declared value is what we need to be looking at, but this also doesn't come without some issues. Let's take it from the top. Property values go through multiple [value stages](https://drafts.csswg.org/css-cascade/#value-stages) before reaching their final value as it will appear when used as part of rendering a page. From start to end, the stages are:

1. [Declared](https://drafts.csswg.org/css-cascade/#declared)
2. [Cascaded](https://drafts.csswg.org/css-cascade/#cascaded)
3. [Specified](https://drafts.csswg.org/css-cascade/#specified)
4. [Computed](https://drafts.csswg.org/css-cascade/#computed)
5. [Used](https://drafts.csswg.org/css-cascade/#used)
6. [Actual](https://drafts.csswg.org/css-cascade/#actual)

As I wrote in https://github.com/act-rules/act-rules.github.io/pull/1251#discussion_r437167976, there can be more than one declared value, but for the rest of the value stages there's at most 1 value for every stage. A small example might help make this clear; consider the following:

```html
<style>
  p {
    line-height: 1.2 !important;
  }
</style>

<p style="line-height: 1.2">Hello world</p>
```

The `<p>` element above has **2** declared values for the `line-height` property: `1.2 !important` and `1.2`. When asked _"is the declared value of the `line-height` property [important](https://drafts.csswg.org/css-cascade/#important)?"_, the answer will of course be _"which declared value?"_, meaning that there's an ambiguity to be resolved. The issue, however, is that the concept of "importance" doesn't really carry over from the declared value as it's sort of resolved as part of cascade. We therefore need to be very specific about _which_ declared value we're referring to if we need to rely on the declared value. I'm guessing we're likely only interested in the one set in the `style` attribute.

Reply:
Wdyt about this - https://github.com/act-rules/act-rules.github.io/pull/1251/files#diff-c8aeb36f7c772db9d9247fd1d52502dfR39